packages/xo-server/src/api/vm.mjs

```diff
@@ -1327,6 +1329,86 @@ importFromEsxi.params = {
   vm: { type: 'string' },
 }
 
+export async function importMultipleFromEsxi({
+  concurrency = 2,
+  host,
+  network,
+  password,
+  sr,
+  sslVerify = true,
+  stopSource = false,
+  stopOnError = false,
+  thin = false,
+  user,
+  vms,
+}) {
+  const task = await this.tasks.create({ name: `importing vms ${vms.join(',')}` })
+  let done = 0
+  return task.run(async () => {
+    Task.set('total', vms.length)
+    Task.set('done', 0)
+    Task.set('progress', 0)
+    const result = {}
+    try {
+      await asyncEach(
+        vms,
+        async vm => {
+          await new Task({ name: `importing vm ${vm}` }).run(async () => {
+            const vmUuid = await this.migrationfromEsxi({
+              host,
+              user,
+              password,
+              sslVerify,
+              thin,
+              vm,
+              sr,
+              network,
+              stopSource,
+            })
+            result[vm] = vmUuid
+          })
+          done++
+          Task.set('done', done)
+          Task.set('progress', Math.round((done * 100) / vms.length))
+        },
+        {
+          concurrency,
+          stopOnError,
+        }
+      )
+      return result
+    } catch (error) {
+      // error is an AggregatedError containing all the errors if stopOnError is false
+      // it is the first error otherwise
+
+      // add the successfull VMs if the UX want to show them something positive
+      error.succeeded = result
+      throw error
+    }
+  })
+}
+
+importMultipleFromEsxi.params = {
+  concurrency: { type: 'number', optional: true },
+  host: { type: 'string' },
+  network: { type: 'string' },
+  password: { type: 'string' },
+  sr: { type: 'string' },
+  sslVerify: { type: 'boolean', optional: true },
+  stopSource: { type: 'boolean', optional: true },
+  stopOnError: { type: 'boolean', optional: true },
+  thin: { type: 'boolean', optional: true },
+  user: { type: 'string' },
+  vms: {
```

Comment:
I still believe that you should move the documentation from the PR descriptions to the JSON-schema directly, this would make them easily accessible when working on the code and also available by API introspection (e.g. `xo-cli`).

For the rest, I stand by my initial comment: documentation in the code or in commits (when it describes the changes), is more visible and more permanent than on a GitHub PR), but it's not a big deal, I'll add them the commit on merge :slightly_smiling_face: